bundles/binding/org.openhab.binding.xbmc/src/main/java/org/openhab/binding/xbmc/rpc/calls/InputBack.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) 2010-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.xbmc.rpc.calls;
+
+import java.util.Collections;
+import java.util.Map;
+
+import org.openhab.binding.xbmc.rpc.RpcCall;
+
+import com.ning.http.client.AsyncHttpClient;
+
+/**
+ * Input.Back RPC
+ *
+ * @author Il Plebs
+ * @since 1.5.0
+ */
+public class InputBack extends RpcCall {
```

Comment:
Hi Ben,
I have just updated the commit.
Can you please check if you can see it correctly?

stefano


On 6 September 2016 at 11:11, Ben Jones <notifications@github.com> wrote:

> In bundles/binding/org.openhab.binding.xbmc/src/main/java/
> org/openhab/binding/xbmc/rpc/calls/InputBack.java
> <https://github.com/openhab/openhab/pull/4622#discussion_r77598406>:
>
> > +package org.openhab.binding.xbmc.rpc.calls;
> > +
> > +import java.util.Collections;
> > +import java.util.Map;
> > +
> > +import org.openhab.binding.xbmc.rpc.RpcCall;
> > +
> > +import com.ning.http.client.AsyncHttpClient;
> > +
> > +/**
> > + * Input.Back RPC
> > + *
> > + * @author Il Plebs
> > + * @since 1.5.0
> > + */
> > +public class InputBack extends RpcCall {
>
> Of course - just whenever you can - thanks for improving the binding!
> … <#m_1691847933481882001_>
> On 06/09/2016 9:09 PM, Plebs wrote: In bundles/binding/org.openhab.
> binding.xbmc/src/main/java/org/openhab/binding/xbmc/rpc/calls/InputBack.java
> <#4622 (comment)
> <https://github.com/openhab/openhab/pull/4622#discussion_r77598220>>: >
> +package org.openhab.binding.xbmc.rpc.calls; > + > +import
> java.util.Collections; > +import java.util.Map; > + > +import
> org.openhab.binding.xbmc.rpc.RpcCall; > + > +import com.ning.http.client.AsyncHttpClient;
> > + > +/** > + * Input.Back RPC > + * > + * @author
> <https://github.com/author> Il Plebs > + * @since
> <https://github.com/since> 1.5.0 > + */ > +public class InputBack extends
> RpcCall { ok. I will out of town for few days. Will fix it next week. Is it
> ok? … <#> On 6 September 2016 at 11:05, Ben Jones ***@***.***> wrote: In
> bundles/binding/org.openhab.binding.xbmc/src/main/java/
> org/openhab/binding/xbmc/rpc/calls/InputBack.java <#4622
> <https://github.com/openhab/openhab/pull/4622> (comment) <#4622 (comment)
> <https://github.com/openhab/openhab/pull/4622#discussion_r77597533>>>: >
> +package org.openhab.binding.xbmc.rpc.calls; > + > +import
> java.util.Collections; > +import java.util.Map; > + > +import
> org.openhab.binding.xbmc.rpc.RpcCall; > + > +import com.ning.http.client.AsyncHttpClient;
> > + > +/** > + * Input.Back RPC > + * > + * @author
> <https://github.com/author> <https://github.com/author> Il Plebs > + *
> @since <https://github.com/since> <https://github.com/since> 1.5.0 > + */
> > +public class InputBack extends RpcCall { I think it is probably worth
> doing it - much simpler code to follow then. … <#m_3632722344795262964_> On
> 06/09/2016 9:03 PM, Plebs wrote: In bundles/binding/org.openhab.
> binding.xbmc/src/main/java/org/openhab/binding/xbmc/rpc/calls/InputBack.java
> <#4622 <https://github.com/openhab/openhab/pull/4622> <#4622
> <https://github.com/openhab/openhab/pull/4622>> (comment) <#4622
> <https://github.com/openhab/openhab/pull/4622> (comment) <#4622 (comment)
> <https://github.com/openhab/openhab/pull/4622#discussion_r77597277>>>>: >
> +package org.openhab.binding.xbmc.rpc.calls; > + > +import
> java.util.Collections; > +import java.util.Map; > + > +import
> org.openhab.binding.xbmc.rpc.RpcCall; > + > +import com.ning.http.client.AsyncHttpClient;
> > + > +/** > + * Input.Back RPC > + * > + * @author
> <https://github.com/author> <https://github.com/author> <
> https://github.com/author> Il Plebs > + * @since
> <https://github.com/since> <https://github.com/since> <
> https://github.com/since> 1.5.0 > + */ > +public class InputBack extends
> RpcCall { Hi Ben, the reason I have added the classes in that way, was to
> maintain the same original logic to reduce the changes. I could probably
> simplify it removing all the 11 Inputxx classes with a single INPUT class
> with several options, like the PlayerGetLabels class. This means that I
> will be adding only 2 new classes: InputCommands and playerGetLabels. What
> do you think? … <#> On 6 September 2016 at 10:37, Ben Jones ***@***.***>
> wrote: In bundles/binding/org.openhab.binding.xbmc/src/main/java/
> org/openhab/binding/xbmc/rpc/calls/InputBack.java <#4622
> <https://github.com/openhab/openhab/pull/4622> <#4622
> <https://github.com/openhab/openhab/pull/4622>> <#4622
> <https://github.com/openhab/openhab/pull/4622> <#4622
> <https://github.com/openhab/openhab/pull/4622>>> (comment) <#4622
> <https://github.com/openhab/openhab/pull/4622> <#4622
> <https://github.com/openhab/openhab/pull/4622>> (comment) <#4622
> <https://github.com/openhab/openhab/pull/4622> (comment) <#4622 (comment)
> <https://github.com/openhab/openhab/pull/4622#discussion_r77593452>>>>>:
> > +package org.openhab.binding.xbmc.rpc.calls; > + > +import
> java.util.Collections; > +import java.util.Map; > + > +import
> org.openhab.binding.xbmc.rpc.RpcCall; > + > +import com.ning.http.client.AsyncHttpClient;
> > + > +/** > + * Input.Back RPC > + * > + * @author
> <https://github.com/author> <https://github.com/author> <
> https://github.com/author> <https://github.com/author> Il Plebs > + *
> @since <https://github.com/since> <https://github.com/since> <
> https://github.com/since> <https://github.com/since> 1.5.0 > + */ >
> +public class InputBack extends RpcCall { Rather than have all these
> classes which are exactly the same apart from the command/name, perhaps it
> would be better to have a single XxxRpcCall class with an enumerator for
> the command? — You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <
> https://github.com/openhab/openhab/pull/4622/files/ 80052fc
> <https://github.com/openhab/openhab/commit/80052fcac5328ee57de2136a17803506032e3a02>
> <https://github.com/openhab/openhab/commit/80052fcac5328ee57de2136a178035
> 06032e3a02>#r77593452>, or mute the thread < https://github.com/
> notifications/unsubscribe-auth/ AUVJ072b2LrHQZiDwp75qhW3YJN98BlAks5qnSZjgaJpZM4J1TtJ>
> . — You are receiving this because you commented. Reply to this email
> directly, view it on GitHub <https://github.com/openhab/
> openhab/pull/4622/files/ 80052fc
> <https://github.com/openhab/openhab/commit/80052fcac5328ee57de2136a17803506032e3a02>
> <https://github.com/openhab/openhab/commit/80052fcac5328ee57de2136a178035
> 06032e3a02>#r77597277>, or mute the thread < https://github.com/
> notifications/unsubscribe-auth/AE046k4u-k8xj6SB_
> 9xf7IqMQgMVIjqnks5qnSxvgaJpZM4J1TtJ>. — You are receiving this because
> you authored the thread. Reply to this email directly, view it on GitHub <
> https://github.com/openhab/openhab/pull/4622/files/
> 80052fcac5328ee57de2136a17803506032e3a02#r77597533>, or mute the thread <
> https://github.com/notifications/unsubscribe-auth/
> AUVJ049cs1O7nDXMIvQC4TcgXzNGTqx2ks5qnSzOgaJpZM4J1TtJ> . — You are
> receiving this because you commented. Reply to this email directly, view it
> on GitHub <https://github.com/openhab/openhab/pull/4622/files/
> 80052fcac5328ee57de2136a17803506032e3a02#r77598220>, or mute the thread <
> https://github.com/notifications/unsubscribe-auth/
> AE046oojsaeQWjTQTglItCcVK6GwhS_Dks5qnS3jgaJpZM4J1TtJ>.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openhab/openhab/pull/4622/files/80052fcac5328ee57de2136a17803506032e3a02#r77598406>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AUVJ01q78CRGO8xuEVJKnF4Nuo06J0e2ks5qnS4ugaJpZM4J1TtJ>
> .
>